### PR TITLE
Make objects not run their code if the room they are in isn't loaded.

### PR DESCRIPTION
**Pull request by arthurtilly**
* Object flag for forcing objects to always be loaded regardless of room
* Does nothing if the object doesn't call bhv_init_room

### CODE REVIEW

**Comment by gheskett** [comment]
Potentially replaces #416

File: src/game/object_helpers.c [hunk 1]
```diff
@@ -1891,15 +1891,15 @@ void bhv_init_room(void) {
     o->oRoom = -1;
 }
 
-void cur_obj_enable_rendering_if_mario_in_room(void) {
-    if (o->oRoom != -1 && gMarioCurrentRoom != 0) {
-        register s32 marioInRoom = (
-            gMarioCurrentRoom == o->oRoom
+u32 is_room_loaded(s32 room) {
```
Comment (gheskett) [inline src/game/object_helpers.c:1894]: Unused parameter

**Review by gheskett — CHANGES_REQUESTED** [review]

File: src/engine/behavior_script.c [hunk 2]
```diff
@@ -820,6 +820,13 @@ void cur_obj_update(void) {
     BhvCommandProc bhvCmdProc;
     s32 bhvProcResult;
 
+    if (!(objFlags & OBJ_FLAG_PROCESS_OUTSIDE_ROOM)) {
+        if (o->oRoom != -1 && gMarioCurrentRoom != 0 && !is_room_loaded(o->oRoom)) {
+            cur_obj_enable_rendering_if_mario_in_room();
```
Comment (gheskett) [inline src/engine/behavior_script.c:825]: Slight nitpick, but it probably makes more sense to copy the three lines out of the `else` block in this function to skip the three unnecessary conditions.

**Review by gheskett — APPROVED** [review]
I assume this has been properly tested to work in-game, and if so then LGTM

**Review by Reonu — APPROVED** [review]
looks good to me i fucked your mom etc

**Review by Reonu — APPROVED** [review]
I like this PR so much I'm approving it again

**Comment by arthurtilly** [comment]
fuck cozies i am based

**Pull request merged**